### DISCUSSION TITLE
I use MilvUS version 2.3, it takes 6-10s to store a piece of data, and it takes about the same time to store batch data, but we have the need for a single new interface. We have a self-developed faq background, when users add their own data, they feel very slow, I would like to ask what method to optimize it, mainly to establish a collection and insert data is relatively slow, but also in the creation of a collection after loading is slow, I can not retrieve the relevant information on the Internet, please help look at it, thank you.

### DISCUSSION THREAD

**Discussion by zhangnn520** [milvus-io/milvus#30422] [Announcement]
I use MilvUS version 2.3, it takes 6-10s to store a piece of data, and it takes about the same time to store batch data, but we have the need for a single new interface. We have a self-developed faq background, when users add their own data, they feel very slow, I would like to ask what method to optimize it, mainly to establish a collection and insert data is relatively slow, but also in the creation of a collection after loading is slow, I can not retrieve the relevant information on the Internet, please help look at it, thank you.

**Comment by zhangnn520** [comment]
![11](https://github.com/milvus-io/milvus/assets/58899025/5a1e89ce-f5f2-45a1-8317-4fb65942a454)

**Reply by zhangnn520** [reply]
![12](https://github.com/milvus-io/milvus/assets/58899025/9f885927-62fc-4b47-9a9b-28ad6618b81a)

**Comment by xiaofan-luan** [comment]
那一部分是milvus的插入耗时呢？

你可以部署一个监控看一下milvus的耗时情况。正常情况下肯定用不了几秒

**Reply by zhangnn520** [reply]
大佬我向您发了一封邮件，如果可以的话，我想加大佬微信，请教大佬。这个问题我也很纳闷：存入单条和存入1000条时间相差不是很多

**Reply by xiaofan-luan** [reply]
建议你部署一下监控，看一下服务端耗时。
然后ping一下server，看一下网络的耗时。
如果是几秒插入一条数据，很大概率不是server的耗时慢

**Comment by zhangnn520** [comment]
我现在插入一条数据大概400ms左右，之前是3-6秒
---- 回复的原邮件 ----
发件人 ***@***.***> 日期 2024年02月01日 13:34 收件人 ***@***.***> 抄送至 ***@***.***>***@***.***> 主题 Re: [milvus-io/milvus] I use MilvUS version 2.3, it takes 6-10s to store a piece of data, and it takes about the same time to store batch data, but we have the need for a single new interface. We have a self-developed faq background, when users add their own data, they feel very slow, I would like to ask what method to optimize it, mainly to establish a collection and insert data is relatively slow, but also in the creation of a collection after loading is slow, I can not retrieve the relevant information on the Internet, please help look at it, thank you. (Discussion #30422) 
建议你部署一下监控，看一下服务端耗时。
然后ping一下server，看一下网络的耗时。
如果是几秒插入一条数据，很大概率不是server的耗时慢

**Reply by xiaofan-luan** [reply]
400ms的话，可能的因素就是网络延迟和消息队列的延迟。
通常我们自己测试到的值应该会比这个短的多

**Comment by zhangnn520** [comment]
是的，我的服务放置在不同服务器，传输会耗费一些时间。大佬，我还有个问题。上述插入一条数据时间是建立在集合和索引已经建好情况下，但是重新建立集合和索引所需时间就非常慢。
---- 回复的原邮件 ----
发件人 ***@***.***> 日期 2024年02月01日 23:33 收件人 ***@***.***> 抄送至 ***@***.***>***@***.***> 主题 Re: [milvus-io/milvus] I use MilvUS version 2.3, it takes 6-10s to store a piece of data, and it takes about the same time to store batch data, but we have the need for a single new interface. We have a self-developed faq background, when users add their own data, they feel very slow, I would like to ask what method to optimize it, mainly to establish a collection and insert data is relatively slow, but also in the creation of a collection after loading is slow, I can not retrieve the relevant information on the Internet, please help look at it, thank you. (Discussion #30422) 
400ms的话，可能的因素就是网络延迟和消息队列的延迟。
通常我们自己测试到的值应该会比这个短的多
